src/text-error-reporter.cpp

```diff
@@ -23,16 +35,42 @@ void text_error_reporter::set_source(padded_string_view input,
 void text_error_reporter::report_impl(error_type type, void *error) {
   QLJS_ASSERT(this->file_path_);
   QLJS_ASSERT(this->locator_.has_value());
+
+  bool format_escape_errors = false;
+
+  switch (this->escape_errors_) {
+  case escape_errors::auto_:
+#if defined(_WIN32)
+    if (this->output_.rdbuf() == std::cerr.rdbuf() && _isatty(_fileno(stderr)))
+#else
+    if (this->output_.rdbuf() == std::cerr.rdbuf() && isatty(fileno(stderr)))
+#endif
+      format_escape_errors = true;
+    else
+      format_escape_errors = false;
+    break;
+  case escape_errors::always:
+    format_escape_errors = true;
+    break;
+  case escape_errors::never:
+    format_escape_errors = false;
+    break;
+  }
```

Comment:
Must fix: This is very expensive to do every time `report` is called. Do this work in the constructor.